dist/stat_data.py

```diff
@@ -824,6 +824,7 @@ def __init__(self, name, desc, flags=''):
     BtreeStat('btree_compact_pages_rewritten_expected', 'btree expected number of compact pages rewritten', 'no_clear,no_scale'),
     BtreeStat('btree_compact_pages_skipped', 'btree compact pages skipped', 'no_clear,no_scale'),
     BtreeStat('btree_compact_skipped', 'btree skipped by compaction as process would not reduce size', 'no_clear,no_scale'),
+    BtreeStat('btree_compact_throttled', 'btree compact was throttled and assigned to perform cache eviction', 'no_clear,no_scale'),
```

Comment:
I think this should be a connection-level session statistic. Grouped with the other compact stats in the category `SessionOpStat`